src/librustc_mir/borrow_check/nll/constraint_generation.rs

```diff
@@ -79,7 +79,8 @@ impl<'cx, 'cg, 'gcx, 'tcx> ConstraintGeneration<'cx, 'cg, 'gcx, 'tcx> {
                 .simulate_block(self.mir, bb, |location, live_locals| {
                     for live_local in live_locals.iter() {
                         let live_local_ty = self.mir.local_decls[live_local].ty;
-                        self.add_regular_live_constraint(live_local_ty, location);
+                        let cause = Cause::Live(live_local, location);
+                        self.add_regular_live_constraint(live_local_ty, location, cause);
```

Comment:
Nit: I think I would "encapsulate" `cause` within region inference, and just modify the signature of `add_regular_live_constraint` to take a `Ty<'tcx>`, `Local`, and `Location`.